lib/Backend/GlobOpt.cpp

```diff
@@ -8909,21 +8910,30 @@ GlobOpt::TypeSpecialization(
             {
                 return instr;
             }
+
+            // need to run typespec if cannot const fold InlineMathAbs, or dst will be dead
+            if (instr->m_opcode != Js::OpCode::InlineMathAbs)
+            {
+                runTypeSpec = false;
```

Comment:
No, but we should bailout at runtime in this case.  There should be a bailout for this situation for when the source isn't a constant, but a variable equal to MinInt.